statemachine/src/main/java/com/amazonaws/stepfunctions/cloudformation/statemachine/DefinitionProcessor.java

```diff
@@ -79,20 +79,21 @@ private static int getNumDefinitionsInModel(final ResourceModel model) {
      * @param metricsRecorder The MetricsRecorder object used for collecting anonymous property usage metrics
      */
     public static void processDefinition(final AmazonWebServicesClientProxy proxy, final ResourceModel model, final MetricsRecorder metricsRecorder) {
```

Comment:
Thought (general, blocking): are you able to fill out the PR description with an overview of the changes, why we're making them, any related GitHub issues, and what testing was done to validate these changes? Did we actually perform end to end testing and validate that the resource is created in Step Functions as expected?

Reply:
I'll do that!

Reply:
I performed end to end testing and validated all cases are created as expected. I'll update this in the PR description and also confirm that customer's GitHub issue is unblocked.

Reply:
json definition DefinitionSubstition also works